programs/pkey/rsa_genkey.c

```diff
@@ -70,6 +73,14 @@ int main( void )
     FILE *fpriv = NULL;
     const char *pers = "rsa_genkey";
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n", ret );
+        goto exit;
```

Comment:
This should be changed to `return ret`. If you go to exit here, you will free uninitialized resources, which might lead to undefined behavior, depending on implementation.